frontend/js/tests/user/Dashboard.test.tsx

```diff
@@ -50,773 +48,763 @@ const props: ListensProps = {
+      within(listenToDelete!).queryByRole("menuitem", {
+        name: "Delete Listen",
+      })
+    );
+
+    expect(await screen.findAllByTestId("listen")).toHaveLength(25);
+  });
+
+  it("does not render delete button if user is not logged in", async () => {
+    renderWithProviders(
+      <Listens />,
+      {
+        currentUser: {
+          name: "foobar",
+        },
+      },
+      {
+        wrapper: reactQueryWrapper,
+      }
+    );
+
+    await waitFor(() => {
+      // Wait for data to be successfully loaded
+      const state = queryClient.getQueryState(queryKey);
+      expect(state?.status === "success").toBeTruthy();
+    });
+
+    const deleteButton = screen.queryAllByRole("menuitem", {
+      name: "Delete Listen",
+    });
+    expect(deleteButton).toHaveLength(0);
+  });
+
+  it("does nothing if the user has no auth token", async () => {
+    renderWithProviders(
+      <Listens />,
+      {
+        currentUser: {
+          name: "iliekcomputers",
+          auth_token: undefined,
+        },
+      },
+      {
+        wrapper: reactQueryWrapper,
+      }
+    );
+
+    await waitFor(() => {
+      // Wait for data to be successfully loaded
+      const state = queryClient.getQueryState(queryKey);
+      expect(state?.status === "success").toBeTruthy();
+    });
+
+    const listensContainer = await screen.findByTestId("listens");
+    const listenCards = await within(listensContainer).findAllByTestId(
+      "listen"
+    );
+    expect(listenCards).toHaveLength(25);
+    const listenToDelete = listenCards[0];
+
+    const deleteButton = within(listenToDelete).getByRole("menuitem", {
+      name: "Delete Listen",
+    });
+    await userEvent.click(deleteButton);
+
+    expect(listenCards).toHaveLength(25);
+  });
+
+  it("doesn't call removeListenFromListenList or update state if status code is not 200", async () => {
+    server.use(
+      http.post("/1/delete-listen", async (path) => {
+        return HttpResponse.json({}, { status: 500 });
+      })
+    );
+    renderWithProviders(
+      <Listens />,
+      {
+        currentUser,
+      },
+      {
+        wrapper: reactQueryWrapper,
       }
     );
+
+    await waitFor(() => {
+      // Wait for data to be successfully loaded
+      const state = queryClient.getQueryState(queryKey);
+      expect(state?.status === "success").toBeTruthy();
+    });
+
+    const listensContainer = await screen.findByTestId("listens");
+    const listenCards = await within(listensContainer).findAllByTestId(
+      "listen"
+    );
+    expect(listenCards).toHaveLength(25);
+    const listenToDelete = listenCards[0];
+
+    const deleteButton = within(listenToDelete).getByRole("menuitem", {
+      name: "Delete Listen",
+    });
+    await userEvent.click(deleteButton);
+
+    expect(listenCards).toHaveLength(25);
+
+    await waitFor(
+      () => {
+        expect(listenCards).toHaveLength(25);
+      },
+      { timeout: 1000 }
+    );
```

Comment:
The expect condition here is doubled. I assume a single one should do the trick, otherwise it might indicate another issue

Reply:
Makes sense, I'll fix this one